### PR TITLE
Set opaque methods as final

### PR DESCRIPTION
**Pull request by allredj**
Setting opaque methods as final by default prevents from erroneously assuming that they can be overridden.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: f2844d8440ac3a995626cbc9cecb459eedf08e7a).
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by peterschrammel — APPROVED** [review]

**Comment by allredj** [comment]
test-gen bump: https://github.com/diffblue/test-gen/pull/2242

**Review by mgudemann — APPROVED** [review]
LGTM

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 1]
```diff
@@ -2210,6 +2210,7 @@ void java_bytecode_convert_methodt::convert_invoke(
                          id2string(symbol.base_name) + "()";
     symbol.type = arg0.type();
     symbol.type.set(ID_access, ID_private);
+    to_java_method_type(symbol.type).set_is_final(true);
```
Comment (antlechner) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:2214]: A comment that repeats the commit message could be added for this (e.g. just above, together with the comments about accessibility).

File: jbmc/unit/java_bytecode/java_bytecode_convert_method/convert_method.cpp [hunk 2]
```diff
@@ -89,5 +89,16 @@ SCENARIO(
         REQUIRE(!function_type.get_is_final());
       }
     }
+    WHEN("When parsing an opaque method")
+    {
+      const symbolt function_symbol =
+        symbol_table.lookup_ref("java::OpaqueClass.staticFunc:()I");
+      const java_method_typet &function_type =
+        require_type::require_java_method(function_symbol.type);
+      THEN("The method should not be marked as final")
```
Comment (antlechner) [inline jbmc/unit/java_bytecode/java_bytecode_convert_method/convert_method.cpp]: Shouldn't this be "The method should be marked as final"?

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 52600973e4c8b454603baf02f954e125917d82fc).
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp (hunk 1 above)
Comment (allredj) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:2214]: Makes sense. Added comment in function header.

File: jbmc/unit/java_bytecode/java_bytecode_convert_method/convert_method.cpp (hunk 2 above)
Comment (allredj) [reply jbmc/unit/java_bytecode/java_bytecode_convert_method/convert_method.cpp]: Oops. Indeed!

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 3]
```diff
@@ -2197,6 +2197,8 @@ void java_bytecode_convert_methodt::convert_invoke(
   // - The translated method could be an inherited protected method, hence
   //   accessible from the original caller, but not from the generated test.
   //   Therefore we must assume that the method is not accessible.
+  // We set opaque methods as final by default to prevent from erroneously
```
Comment (antlechner) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: :pick: "to prevent from" -> "to avoid" / "to prevent jbmc from"

**Review by antlechner — APPROVED** [review]

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp (hunk 3 above)
Comment (allredj) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp]: changed

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 9ce743c04e77b8abdedf855bded3ab63748e8f50).
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 876fb6e983916d63501ca2bc6dd98fa31810e4c2).
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Pull request merged**